docs/source/contributor-guide/plugin_overview.md

```diff
@@ -17,30 +17,37 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-# Comet Plugin Overview
+# Comet Plugin Architecture
 
-The entry point to Comet is the `org.apache.spark.CometPlugin` class, which can be registered with Spark by adding the following setting to the Spark configuration when launching `spark-shell` or `spark-submit`:
+## Comet SQL Plugin
+
+The entry point to Comet is the `org.apache.spark.CometPlugin` class, which can be registered with Spark by adding the 
+following setting to the Spark configuration when launching `spark-shell` or `spark-submit`:
 
 ```
 --conf spark.plugins=org.apache.spark.CometPlugin
 ```
 
-On initialization, this class registers two physical plan optimization rules with Spark: `CometScanRule` and `CometExecRule`. These rules run whenever a query stage is being planned.
+On initialization, this class registers two physical plan optimization rules with Spark: `CometScanRule` 
+and `CometExecRule`. These rules run whenever a query stage is being planned during Adaptive Query Execution.
```

Comment:
does that mean CometExecRule is not called if AQE is disabled?

Reply:
No, I should clarify that